src/ddscxx/include/org/eclipse/cyclonedds/topic/cdr_dummys.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright(c) 2021 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#ifndef CDR_DUMMYS_HPP_
+#define CDR_DUMMYS_HPP_
+
+#include <string>
+#include <vector>
+#include <array>
+
+template<size_t N>
+class idl_string : public std::string {
+public:
+  idl_string() : std::string() {}
```

Comment:
Maybe an even more elegant way is to inherit constructors here?

`using std::string::string;` in stead of the explicit declarations?

Reply:
I'm not the worlds foremost C++ expert, but if the language allows you to do that, then an alias seems like the right choice.